jctools-core/src/main/java/org/jctools/queues/MpmcUnboundedXaddArrayQueue.java

```diff
@@ -216,6 +216,14 @@ private E linkNextConsumerChunkAndPoll(
         MpmcUnboundedXaddChunk<E> next,
         long expectedChunkIndex)
     {
+        if (next == null) {
+            final long cChunkIndex = expectedChunkIndex - 1;
```

Comment:
Taken both the good advices, thanks :)

I've changed back the mpsc method name instead because the fact that `spinForNextIfNotEmpty` is trying append a new chunk is just an implementation detail and I'm not sure it should leak in the method name